apps/promotion-app/backend/modules/people/hr.bal

```diff
@@ -39,34 +40,97 @@ public isolated function fetchEmployeesBasicInfo(string workEmail) returns Emplo
     return response.data.employee;
 }
 
-# Retrieves employees.
+# Retrieve the Probational Employee data.
 #
-# + return - Employee Info Array
-public isolated function getEmployees() returns EmployeeBasic[]|error {
-
-    EmployeeFilter filter = {
-        employeeStatus: [Active, Marked\ leaver],
-        employmentType: [PERMANENT, CONSULTANCY, PART\ TIME\ CONSULTANCY]
-    };
-
-    string document = string `query getAllEmployees($filter: EmployeeFilter!, $limit: Int, $offset: Int) {
-        employees(filter: $filter, limit: $limit, offset: $offset) {
-            workEmail
-            firstName
-            lastName
+# + workEmail - WSO2 Email
+# + return - Employee Info
+public isolated function getEmployee(string workEmail) returns Employee|error {
+
+    string document = string `
+        query employeeQuery ($workEmail: String!) {
+            employee(email: $workEmail) {
+                employeeId,
+                firstName,
+                lastName,
+                lead,
+                managerEmail,
+                reportsTo,
+                jobBand,
+                jobRole,
+                lastPromotedDate,
+                startDate,
+                team,
+                employeeStatus,
+                businessUnit,
+                department,
+                team,
+                subTeam,
+                employeeThumbnail
+            }
+        }
+    `;
+
+    EmployeeResults|error employeeData = hrClient->execute(document, {workEmail});
+
+    if employeeData is error {
+        log:printError(employeeData.toString());
+        return error("Employee data retrieving service error!");
```

Comment:
Is this error message correct? I don't think `service error` is adding any value.